### DISCUSSION TITLE
How to contribute to documentation

### DISCUSSION THREAD

**Discussion by ClaytonCurtis1947** [freeplane/docs#120] [General]
The docs page "How to Start Contributing" states that "You can start a discussion post (in the docs project) saying you want to contribute to the documentation and the Freeplane team will respond and assist you."

I'd like to see if I can contribute in a small way to the community by working on documentation.
Guidance would be appreciated.

PS: The directions quoted do not specify -where- in the docs project one should start a post. I hope this is the right place.

**Comment by macmarrum** [comment]
Hi @ClaytonCurtis1947 

Welcome!

Here is the usual way to contribute
1. Fork `freeplane/docs` so that you have its copy as your own repository
2. Make changes in your forked repository, commit and push
4. Create a Pull Request (to merge your changes into `freeplane/docs`)
5. Mention the original authors in your Pull Request, as described in 
    - https://github.com/freeplane/docs/discussions/62

When in doubt, ask :smile:

**Reply by kenfhill84083** [reply]
> Hi @ClaytonCurtis1947
> 
> Welcome!
> 
> Here is the usual way to contribute
> 
>     1. Fork `freeplane/docs` so that you have its copy as your own repository
> 
>     2. Make changes in your forked repository, commit and push
> 
>     3. Create a Pull Request (to merge your changes into `freeplane/docs`)
> 
>     4. Mention the original authors in your Pull Request, as described in
>        
>        * [How previous Authors should be contacted in case of changing existing docu pages #62](https://github.com/freeplane/docs/discussions/62)
> 
> 
> When in doubt, ask 😄

I am taking  you up on your advice because I am in major doubt.  I am so confused.  I forked Freeplane/docs but I feel like I've stepped into Coder Land - Push, Commit, Pull Request, .  I can' find any of the documents I saw in https://docs.freeplane.org/user-documentation/user-documentation.html.  And the instructions about mdbook make no sense to me.

What happens if I just click on the link at the bottom of the documentation pages - "Found a mistake? Edit this page on GitHub"?  Would I be going rogue at that point?

I had an expectation that each feature would have a template of some kind - Purpose, How to, Examples ...  Is there anything like that?  (Not that it would do me any good at this point because I don't see how to actually write text anywhere.

At this point I feel I can best contribute to the general body of informal online Freeplane documentation by continuing to develop my Youtube channel.  But that does little toward an official, cohesive documentation.

Thanks.

Sorry if I should have posted this elsewhere.

**Reply by macmarrum** [reply]
@kenfhill84083 

> I can' find any of the documents I saw in https://docs.freeplane.org/user-documentation/user-documentation.html.

The Markdown (.md) documents making up user-documentation can be found in https://github.com/freeplane/docs/tree/main/src/docs/user-documentation

> And the instructions about mdbook make no sense to me.

mdbook instructions are only if someone wants to run the tool locally (on their computer) to convert .md files into .html files. It's fine to skip mdbook.

> What happens if I just click on the link at the bottom of the documentation pages - "Found a mistake? Edit this page on GitHub"? 

Please feel free to use "Found a mistake? Edit this page on GitHub".
Since this way skips the Pull Request steps, it would be great if, in case it is needed, you could contact the original author(s) e.g. by creating a discussion post
- #62 

> I had an expectation that each feature would have a template of some kind - Purpose, How to, Examples ... Is there anything like that? 

I think there aren't any templates.

> At this point I feel I can best contribute to the general body of informal online Freeplane documentation by continuing to develop my Youtube channel. But that does little toward an official, cohesive documentation.

Whatever you do to contribute, it's much appreciated :+1: :smile:

**Comment by kenfhill84083** [comment]
I noticed a slight inaccuracy on the "understanding nodes" page for "connectors"  https://docs.freeplane.org/attic/new-wiki-initiative/understanding-nodes.html#connectors

It reads "Unlike edges, connectors must be added manually, one by one. To add a connector, select **two** nodes,"

But in my experience, you can select **more than 2 nodes** before pressing Ctrl+L.

My question is: Can/should I just edit that (with the link at the page bottom) or is there a better way - to avoid being premature, in case discussion is called for?

**Reply by macmarrum** [reply]
A good catch :+1: 

> Can/should I just edit that (with the link at the page bottom) or is there a better way - to avoid being premature, in case discussion is called for?

I believe this is covered in
- #62

i.e. you find out who added the content, create a PR and mention them by name (e.g. @kenfhill84083) to get their opinion. Or use some other means of direct communication, if available.

Here's how to find out in GitHub who added the content:
1. Find the .md file in GitHub repository
2. Click the **Blame** button
3. Find the content you are interested in
4. Assess if the change was an addition of the content; you can view the Pull Request (if available - in a tooltip/popup) that the change was introduced with
5. If not, use the button **View blame prior to this change** and repeat steps 3, 4, 5


![GitHub-blame-deep-dive](https://github.com/freeplane/docs/assets/7197131/6d246654-82df-4666-accd-540074256e1a)

After a few steps (View blame prior to this change)

![GitHub-blame-finally](https://github.com/freeplane/docs/assets/7197131/b3317cb0-ab7a-42b4-bb49-d5f8b7220a5a)

----
----

In this case it's my content, added on Dec 18, 2021 and changed a few times.
I propose that I'll correct it in a PR which you'll be able to review and provide your comments.
